opendcs-rest-api/src/test/java/org/opendcs/odcsapi/res/it/AppResourcesIT.java

```diff
@@ -0,0 +1,474 @@
+
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@Tag("integration")
+@ExtendWith(DatabaseContextProvider.class)
+final class AppResourcesIT extends BaseIT
+{
+	private static Long appid;
+	private static SessionFilter sessionFilter;
+
+	@BeforeEach
+	void setUp() throws Exception
+	{
+		setUpCreds();
+		sessionFilter = new SessionFilter();
+		authenticate(sessionFilter);
+
+		String appJson = getJsonFromResource("app_insert_data.json");
+
+		ExtractableResponse<Response> response = given()
+			.log().ifValidationFails(LogDetail.ALL, true)
+			.accept(MediaType.APPLICATION_JSON)
+			.contentType(MediaType.APPLICATION_JSON)
+			.header("Authorization", authHeader)
+			.body(appJson)
+			.filter(sessionFilter)
+		.when()
+			.redirects().follow(true)
+			.redirects().max(3)
+			.post("app")
+		.then()
+			.log().ifValidationFails(LogDetail.ALL, true)
+		.assertThat()
+			.statusCode(is(HttpServletResponse.SC_OK))
+			.extract()
+		;
+
+		appid = response.body().jsonPath().getLong("appId");
+	}
+
+	@AfterEach
+	void tearDown()
+	{
+		given()
+			.log().ifValidationFails(LogDetail.ALL, true)
+			.accept(MediaType.APPLICATION_JSON)
+			.queryParam("appid", appid)
+			.header("Authorization", authHeader)
+			.filter(sessionFilter)
+		.when()
+			.redirects().follow(true)
+			.redirects().max(3)
+			.delete("app")
+		.then()
+			.log().ifValidationFails(LogDetail.ALL, true)
+		.assertThat()
+			.statusCode(is(HttpServletResponse.SC_OK))
+		;
+
+		logout(sessionFilter);
+	}
+
+	@TestTemplate
+	void testGetAppRefs()
+	{
+		JsonPath expected = getJsonPathFromResource("app_get_refs_expected.json");
+
+		ExtractableResponse<Response> response = given()
+			.log().ifValidationFails(LogDetail.ALL, true)
+			.accept(MediaType.APPLICATION_JSON)
+			.header("Authorization", authHeader)
+			.filter(sessionFilter)
+		.when()
+			.redirects().follow(true)
+			.redirects().max(3)
+			.get("apprefs")
+		.then()
+			.log().ifValidationFails(LogDetail.ALL, true)
+		.assertThat()
+			.statusCode(is(HttpServletResponse.SC_OK))
+			.extract()
+		;
+
+		JsonPath actual = response.body().jsonPath();
+
+		boolean found = false;
+		for (int i = 0; i < actual.getList("").size(); i++)
+		{
+			String actualAppName = actual.get("[" + i + "].appName");
+			String expectedAppName = expected.get("[" + i + "].appName");
+			if (actualAppName.equalsIgnoreCase(expectedAppName))
+			{
+				assertEquals(actualAppName, expectedAppName);
+				assertEquals(actual.get("[" + i + "].appType").toString(),
+						expected.get("[" + i + "].appType").toString());
+				assertEquals(actual.get("[" + i + "].comment").toString(),
```

Comment:
All this String manipulation isn't very elegant.  Can't you hang on to the list from the getList("") call and do normal List stuff with that?

Reply:
Cleaned up to remove String manipulation